tests/SerdeTest.php

```diff
@@ -607,6 +610,42 @@ protected function nested_objects_with_flattening_validate(mixed $serialized): v
 
     }
 
+    /**
+    * @test
+    * @group typemap
+    */
+    public function nested_objects_with_flattening2(): void
+    {
+        foreach ($this->formatters as $formatter) {
+            if (($formatter->format() === $this->format) && !$formatter instanceof SupportsCollecting) {
+                $this->markTestSkipped('Skipping flattening tests on non-flattening formatters');
+            }
+        }
+
+        $s = new SerdeCommon(formatters: $this->formatters);
+
+        $data = new ObjectWithFlattenedNestedFlattenObject('First', null);
+
+        $serialized = $s->serialize($data, $this->format);
+
+        $this->nested_objects_with_flattening2_validate($serialized);
+
+        $result = $s->deserialize($serialized, from: $this->format, to: ObjectWithFlattenedNestedFlattenObject::class);
+
+        // deserialization is asymmetric for now (the nested object is created even in cases when it is nullable and null in deserialized data)
```

Comment:
That seems undesireable.  If there's an explicit null value, why would it create a a new object?  That means something different.